pkg/build/builder/dockerutil.go

```diff
@@ -40,8 +40,13 @@ func buildImage(client DockerClient, dir string, noCache bool, tag string, tar t
 		return err
 	}
 	defer tarStream.Close()
+	var auth docker.AuthConfigurations
+	if pullAuth != nil {
+		auth = *pullAuth
+	}
 	opts := docker.BuildImageOptions{
 		Name:           tag,
+		AuthConfigs:    auth,
```

Comment:
yeah, I'm doing some other changes now so I will include this as well. good catch!